internal/util.go

```diff
@@ -64,4 +82,63 @@ func ExtractDomainServiceFromServiceAccount(saName string) (string, string) {
 		service = saName[idx+1:]
 	}
 	return domain, service
-}
\ No newline at end of file
+}
+
+func ExtractSpiffeURIFromCSR(csrBytes []byte) (string, error) {
+	// Decode the PEM encoded CSR
+	block, rest := pem.Decode(csrBytes)
+	if block == nil {
+		return "", fmt.Errorf("no PEM block found in input")
+	}
+	if block.Type != "CERTIFICATE REQUEST" {
+		return "", fmt.Errorf("not a certificate request PEM block")
+	}
+
+	if len(rest) > 0 {
+		return "", fmt.Errorf("unexpected data found after PEM block")
+	}
+
+	// Parse the CSR
+	csr, err := x509.ParseCertificateRequest(block.Bytes)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse CSR: %v", err)
+	}
+
+	// Extract the URI SAN from the CSR
+	for _, ext := range csr.Extensions {
```

Comment:
Is there any reason why to look at the extensions using asn1 instead of just accessing the already parsed fields directly?